src/figuresClasses.ts

```diff
@@ -1,19 +1,91 @@
-export interface Figure {
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
+// export enum Color {
+//   red = 'red',
+//   green = 'green',
+//   blue = 'green',
+// }
 
+type Color = 'red' | 'green' | 'blue';
+
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const sides: number[] = [a, b, c];
+    const perimeter: number = a + b + c;
 
+    if (sides.some((side: number) => side <= 0)) {
+      throw new Error('The sides values must be greater than 0');
+    }
```

Comment:
you checked input values for each figure
will be better to create separate function for this purpose